backend/main.go

```diff
@@ -138,78 +200,144 @@ func roll(level int, rolls int) {
 			}
 
 			if c, ok := rollResults[champion]; ok {
-				rollResults[champion] = rollResult{champion, cost + 1, c.Appearances + 1}
+				rollResults[champion] = rollResult{champion, cost + 1, c.Appearances + 1, ""}
 			} else {
-				rollResults[champion] = rollResult{champion, cost + 1, 1}
+				rollResults[champion] = rollResult{champion, cost + 1, 1, ""}
 			}
 		}
 	}
+
+	// calculate the percentage of appearance for each champion
+	for i, s := range rollResults {
+		p := (float32(s.Appearances) * 100) / (float32(rolls) * 5)
+		s.Percentage = fmt.Sprintf("%.2f", p) + "%"
```

Comment:
Convert values to strings when you are presenting them - Converting them early results in a loss of information